### PR TITLE
Hacktoberfest 2020: Added computer vision algorithm

### PR DESCRIPTION
**Pull request by TanayKarve**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by sharpblade4** [comment]
@TanayKarve you can optimize it in magnitudes by using numpy vectorian oprerations instead of python for loops

**Comment by TanayKarve** [comment]
@sharpblade4 yes i was planning to do it, however wanted to write this code in pure python, thus avoiding any dependencies.

**Review by TanayKarve — COMMENTED** [review]
avoiding numpy to make this implement in pure python.

**Comment by TanayKarve** [comment]
@sharpblade4 sorry to disturb you, but its been a week, can you please accept this PR?

**Comment by sharpblade4** [comment]
@TanayKarve  sorry for confusing you, I'm just another contributer - not an member of TheAlgorithms/Python.

File: computer_vision/meanthreshold.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    image: is a grayscale PIL image object
+    """
+    height, width = image.size
+    mean = 0
+    pixels = image.load()
+    for i in range(width):
+        for j in range(height):
+            pixel = pixels[j, i]
+            mean += pixel
+    mean //= width * height
```
Comment (cclauss) [inline computer_vision/meanthreshold.py:20]:
```suggestion
    total = sum(sum(pixel for pixel in row) for row in image.load())
    mean = total // (width * height)
```

File: computer_vision/meanthreshold.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    for j in range(width):
+        for i in range(height):
+            if pixels[i, j] > mean:
+                pixels[i, j] = 255
+            else:
+                pixels[i, j] = 0
```
Comment (cclauss) [inline computer_vision/meanthreshold.py:27]:
```suggestion
            pixels[i, j] = 255 if pixels[i, j] > mean else 0
```

File: computer_vision/meanthreshold.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    return image
+
+
+if __name__ == "__main__":
+    image = Image.open("path_to_image").convert("L")
+    image = mean_threshold(image)
```
Comment (cclauss) [inline computer_vision/meanthreshold.py:33]:
```suggestion
    image = mean_threshold(Image.open("path_to_image").convert("L"))
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: computer_vision/meanthreshold.py (hunk 1 above)
Comment (TanayKarve) [reply computer_vision/meanthreshold.py:20]: Yes I tried using list builder comprehension initially. However `image.load()` returns a `PixelAccess` object which according to the [documentation](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.load) is not [Iterable](https://stackoverflow.com/a/1111950). Thus when running this code the following exception is thrown:
`TypeError: 'PixelAccess' object is not iterable`

**Review by cclauss — APPROVED** [review]
Well done!  Thanks.

**Pull request merged**